### PR TITLE
New coding standard

### PR DESCRIPTION
**Pull request by Koopzington**
Usage of the newly introduced zendframework/zend-coding-standard for CS checks.

### CODE REVIEW

**Review by froschdesign — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -37,7 +35,9 @@
         "zendframework/zend-session": "^2.6.2",
         "zendframework/zend-mvc-plugin-flashmessenger": "^0.1.0",
         "zendframework/zend-mvc-console": "^1.1.8",
-        "zendframework/zend-validator": "^2.8"
+        "zendframework/zend-validator": "^2.8",
+        "zendframework/zend-coding-standard": "~1.0.0",
+        "phpunit/phpunit": "^4.6 || ^5.0"
```
Comment (samsonasik) [inline composer.json]: phpunit should in require path to manage syntax compat, also, use phpunit ^5.6 || ^6.0 in favor of https://github.com/zendframework/zend-test/pull/43

**Review by samsonasik — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (samsonasik) [reply composer.json]: I meant, phpunit ^5.7 || ^6.0

File: composer.json [hunk 2]
```diff
@@ -37,7 +36,8 @@
         "zendframework/zend-mvc-plugin-flashmessenger": "^0.1.0",
         "zendframework/zend-mvc-console": "^1.1.8",
         "zendframework/zend-validator": "^2.8",
-        "zendframework/zend-coding-standard": "~1.0.0"
+        "zendframework/zend-coding-standard": "~1.0.0",
+        "phpunit/phpunit": "^4.6 || ^5.0"
```
Comment (froschdesign) [inline composer.json]: PHPUnit is required in this component,

**Review by froschdesign — CHANGES_REQUESTED** [review]

**Comment by samsonasik** [comment]
move phpunit to "require" section in composer.json, also, this should belongs to develop branch, after it pointed to develop, please rebase agains develop.

**Comment by Koopzington** [comment]
https://github.com/zendframework/zend-mail/pull/121
every other PR regarding the coding standard was against master, why should we do it differently in this case?

**Comment by samsonasik** [comment]
upping phpunit was went to develop, and I think goes to master without coding fix will go bugs, please check #43 for the phpunit 5 & 6 compat fix

**Comment by Koopzington** [comment]
not entirely sure but i think i just screwed up, didn't i?

**Comment by samsonasik** [comment]
You can do with the following steps:

```
git checkout develop
git pull git@github.com:zendframework/zend-test.git develop
git checkout new-coding-standard
git rebase develop
```

at this stage, you can fixes the conflicts (don't commit), then do

```
git add .
git rebase --continue
```

Next, do force push:

```
git push --force origin new-coding-standard
```

Lastly, point the PR to develop branch via "Edit" button in PR title, select "base" branch to develop.

**Comment by Koopzington** [comment]
seems to look good now

File: .travis.yml [hunk 3]
```diff
@@ -31,13 +30,12 @@ matrix:
...
       env:
-        - EXECUTE_TEST_COVERALLS=true
+        - TEST_COVERAGE=true
         - DEPS=locked
         - DEPLOY_DOCS="$(if [[ $TRAVIS_BRANCH == 'master' && $TRAVIS_PULL_REQUEST == 'false' ]]; then echo -n 'true' ; else echo -n 'false' ; fi)"
         - PATH="$HOME/.local/bin:$PATH"
     - php: 5.6
       env:
-        - PHPUNIT_VERSION=~5.0
         - DEPS=latest
```
Comment (samsonasik) [inline .travis.yml]: remove this php: 5.6 env with deps "latest" entry:
```
    - php: 5.6
      env:
        - DEPS=latest
```
as php 5.6 can't install phpunit 6

Comment (samsonasik) [reply .travis.yml]: check modified .travis at https://raw.githubusercontent.com/samsonasik/zend-test/0b0bf0f351e228557997edac539352a1d5932be5/.travis.yml that applied at PR #43

File: .travis.yml [hunk 4]
```diff
@@ -65,10 +60,9 @@ notifications:
   email: false
 
 before_install:
-  - if [[ $EXECUTE_TEST_COVERALLS != 'true' ]]; then phpenv config-rm xdebug.ini || return 0 ; fi
+  - if [[ $TEST_COVERAGE != 'true' ]]; then phpenv config-rm xdebug.ini || return 0 ; fi
   - travis_retry composer self-update
-  - if [[ $EXECUTE_TEST_COVERALLS == 'true' ]]; then composer require --dev --no-update satooshi/php-coveralls ; fi
-  - composer require --no-update "phpunit/phpunit:$PHPUNIT_VERSION"
+  - if [[ $TEST_COVERAGE == 'true' ]]; then composer require --dev --no-update satooshi/php-coveralls ; fi
```
Comment (samsonasik) [inline .travis.yml:65]: `--no-update` make it not install anything php-coveralls bin, change with include $COMPOSER_ARGS and remove the `--no-update` like the following :

```yaml
- if [[ $TEST_COVERAGE == 'true' ]]; then composer require --dev satooshi/php-coveralls $COMPOSER_ARGS; fi
```

Comment (samsonasik) [reply .travis.yml:65]: see https://github.com/zendframework/zend-test/pull/43/commits/226100a02e231dc12fe0f8e52e97434ba6587531

Comment (samsonasik) [inline .travis.yml]: need to be re-add as @webimpress  suggestion at https://github.com/zendframework/zend-test/pull/43#discussion_r99944201

File: .travis.yml (hunk 3 above)
Comment (samsonasik) [reply .travis.yml]: need to be re-add as @webimpress suggestion at #43 (comment)

File: .travis.yml (hunk 4 above)
Comment (Koopzington) [reply .travis.yml:65]: Can we please stop abusing this PR for changes which weren't intended with it?
See #35

Comment (samsonasik) [reply .travis.yml:65]: I'm sorry for it, just make sure if your PR merged first, the coveralls command will still works, nothing else.

File: composer.json [hunk 5]
```diff
@@ -14,7 +14,7 @@
     },
     "require": {
         "php": "^5.6 || ^7.0",
-        "phpunit/phpunit": "^4.0 || ^5.0 || ^6.0",
+        "phpunit/phpunit": "^5.7 || ^6.0",
```
Comment (weierophinney) [inline composer.json:17]: We need to keep 4.0 support for BC purposes at this point. That version still works with all versions of PHP; however, there's syntax from v4 that doesn't work later, which would mean that upping the PHPUnit version could break existing tests for people using this component.

I'll revert this on merge.

**Pull request merged**